torch/csrc/jit/python/init.cpp

```diff
@@ -1253,9 +1253,27 @@ void initJITBindings(PyObject* module) {
       .def(
           "__str__",
           [](c10::SymNode a) { return a->str(); })
-      .def("__repr__", [](c10::SymNode a) {
-        return a->str();
-      });
+      .def(
+          "__repr__",
+          [](c10::SymNode a) { return a->str(); })
+      .def(
+          "is_constant",
+          [](const c10::SymNode& node){
+            return (
+              node->constant_int().has_value()
+              || node->constant_bool().has_value()
+            );
+          })
+      .def(
+          "is_symbolic",
+          [](const c10::SymNode& node) {
+            return (
+                !node->singleton_int().has_value()
+                && !node->constant_bool().has_value()
+                && !node->constant_int().has_value()
```

Comment:
It would be more symmetric with the other functions to add a new virtual method on SymNode and call that directly to query this information

Reply:
Done.